packages/eslint-plugin/docs/rules/no-unnecessary-type-parameters.mdx

```diff
@@ -0,0 +1,115 @@
+---
+description: 'Disallow type parameters that only appear once.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unnecessary-type-parameters** for documentation.
+
+This rule forbids type parameters that only appear once in a function, method, or class definition.
+
+Type parameters relate two types.
+If a type parameter only appears once, then it is not relating anything.
+It can usually be replaced with explicit types such as `unknown`.
+
+At best unnecessary type parameters make code harder to read.
+At worst they can be used to disguise unsafe type assertions.
+
+:::warning Early Stage
+This rule was recently added to typescript-eslint and still considered experimental.
+It might change significantly between minor versions.
+Please try it out and give us feedback!
+:::
```

Comment:
From chats with @bradzacher and @Josh-Cena: we're roughly ready to merge this rule, but it's still somewhat experimental / something-we-haven't-tried-ourselves-before. So instead of putting in `strict` for now I'll add this notice and file a followup to move to `strict`.

Reply:
(for posterity: the chats were about adding a notice; I felt icky after the fact about both having a notice and leaving it in `strict`)